src/main/java/org/jsmart/zerocode/core/engine/mocker/RestEndPointMocker.java

```diff
@@ -1,51 +1,119 @@
-                                .withHeader("Content-Type", APPLICATION_JSON)
-                                .withBody(jsonBodyRequest)));
+                LOGGER.info("*****WireMock- Mocking the POST endpoint");
+                givenThat(createPostRequestBuilder(mockStep)
+                        .willReturn(responseBuilder(mockStep, jsonBodyRequest)));
+                LOGGER.info("WireMock- Mocking the POST endpoint -done-*****");
             }
             else if("PUT".equals(mockStep.getOperation())){
-                givenThat(WireMock.put(urlEqualTo(mockStep.getUrl()))
-                        .willReturn(aResponse()
-                                .withStatus(mockStep.getResponse().get("status").asInt())
-                                .withHeader("Content-Type", APPLICATION_JSON)
-                                .withBody(jsonBodyRequest)));
+                LOGGER.info("*****WireMock- Mocking the PUT endpoint");
+                givenThat(createPutRequestBuilder(mockStep)
+                        .willReturn(responseBuilder(mockStep, jsonBodyRequest)));
+                LOGGER.info("WireMock- Mocking the PUT endpoint -done-*****");
             }
         });
     }
 
+
+    public static void restartWireMock(int dynamicPort) {
+        wireMockServer = new WireMockServer(wireMockConfig().port(dynamicPort)); // <-- Strange
+        /*
+         * Stop the wireMock server if it is running previously due to any other tests.
+         */
+        wireMockServer.stop();
+        wireMockServer.start();
+        WireMock.configureFor("localhost", dynamicPort); // <-- Repetition of PORT was needed, this is a wireMock bug
+    }
+
+    private static MappingBuilder createPutRequestBuilder(MockStep mockStep) {
+        final MappingBuilder requestBuilder = put(urlEqualTo(mockStep.getUrl()));
+        return createRequestBuilderWithHeaders(mockStep, requestBuilder);
+    }
+
+    private static MappingBuilder createPostRequestBuilder(MockStep mockStep) {
+        final MappingBuilder requestBuilder = post(urlEqualTo(mockStep.getUrl()));
+        return createRequestBuilderWithHeaders(mockStep, requestBuilder);
+    }
+
+    private static MappingBuilder createGetRequestBuilder(MockStep mockStep) {
+        final MappingBuilder requestBuilder = get(urlEqualTo(mockStep.getUrl()));
+        return createRequestBuilderWithHeaders(mockStep, requestBuilder);
+    }
+
+    private static MappingBuilder createRequestBuilderWithHeaders(MockStep mockStep, MappingBuilder requestBuilder) {
+
+        final String bodyJson = mockStep.getBody();
+        // -----------------------------------------------
+        // read request body and set to request builder
+        // -----------------------------------------------
+        if(StringUtils.isNotEmpty(bodyJson)){
+            requestBuilder.withRequestBody(equalToJson(bodyJson));
+        }
+
+        final Map<String, Object> headersMap = mockStep.getHeadersMap();
```

Comment:
Can be optimized via injected mapper. But not a big deal as it's the same mapper used.

Reply:
ok. will take care.